controller/userController.js

```diff
@@ -114,157 +103,169 @@ const create = (req, res) => {
+            newUser.passwordHash = null;
+            newUser.salt = null;
+
+            return res.status(HttpStatus.OK).json({ token, user: newUser });
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                message: 'User not found.'
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            message: 'Internal server error'
+        });
+    });
+};
+
+const changePassword = (req, res) => {
+    const loggedUser = req.decoded;
+
+    if (!ValidationUtil.isValidObjectId(loggedUser._id)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid ID' });
+    }
+
+    let user = {};
     if (req.body.passwordHash) {
         let retVal = getPasswordHash(req.body.passwordHash);
         user.salt = retVal.salt;
         user.passwordHash = retVal.passwordHash;
     }
 
-    User.findOneAndUpdate({ _id: userId }, user, { useFindAndModify: false, new: true, runValidators: true }).then(newUser => {
+    User.findOneAndUpdate({ _id: loggedUser._id }, user, { useFindAndModify: false, new: true, runValidators: true }).then(newUser => {
         if (newUser) {
+            // Generate new JWT
+            const token = jwt.sign(JSON.stringify(newUser), AppConfig.SECRET);
             newUser.passwordHash = null;
             newUser.salt = null;
-            return res.status(HttpStatus.OK).json(newUser);
+
+            return res.status(HttpStatus.OK).json({ token, user: newUser });
         } else {
             return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
                 message: 'User not found.'
             });
         }
     }).catch(err => {
         return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
-            status: StatusEnum['ERROR'],
-            message: ValidationUtil.buildErrorMessage(err, 'update', 'user')
+            message: 'Internal server error'
         });
     });
 };
 
-const remove = (req, res) => {
-    const userId = req.params.id;
-    const loggedUser = req.decoded;
+const changeActiveStatus = (req, res) => {
+    if (!req.params.id || !ValidationUtil.isValidObjectId(req.params.id)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid ID' });
+    }
 
-    if (userId.length != AppConfig.OBJECT_ID_LEN) {
-        return res.status(HttpStatus.BAD_REQUEST).json({
-            status: StatusEnum['ERROR'],
-            message: 'Invalid ID'
-        });
+    if (!req.params.body || !req.params.status) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid status' });
     }
 
-    User.findOneAndUpdate({ _id: userId, companyID: loggedUser.companyID }, { active: 0 }, { useFindAndModify: false, new: true, runValidators: true }).then(user => {
+    const userId = req.params.id;
+    const status = req.params.status;
```

Comment:
I understand sending the id of the user via params, but shouldn't other properties like 'active' prop value be sent via request body?

Reply:
That was the initial implementation, but as @SavoX2 mentioned it would complicate things on front end. 
Generally update parameters should be sen't via body. For me it makes more sense to send the status via body and the id as parameter. Logically it makes more sense to have a unique route for each user which is used to change his active status. 
I don't honestly think that sending one parameter via `param` and the other via `body` will give the developer any confusion

Reply:
https://stackoverflow.com/questions/47921841/rest-api-update-single-field-of-resource look into this, single parameter update practice for boolean values. user/{user_id}/activate with PATCH http request would do the trick and would remove the other link parameter.

Reply:
@zirlman @SavoX2  What's the verdict on this? I'm for the option to send the update values via request body, and only the endpoint object ID's should be present in the request URL.

Reply:
@srkacha  I agree with you.